File: simplest/which.py
import os
import sys

dirs  = (os.environ['PATH']).split(';')
clean_dirs = filter(lambda dir: dir != '', dirs)
for dir in clean_dirs :
    expected_path = dir + "\\" + sys.argv[1]  
    if os.path.isfile(expected_path) :
        print 'found at ' + expected_path
        exit()

print 'command not found'

File: simplest/which.cs
using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;

namespace CmdUtils
{
    class which
    {
        static void Main(string[] args)
        {
            if (args.Length != 1)
            {
                Console.WriteLine("no command specified");
                return;
            }

            string command = args[0];

            var path_dirs = Environment.GetEnvironmentVariable("PATH")
                .Split(';')
                .Where(dir => dir != null && dir != "")
                .ToList();

            var first = path_dirs
                .FirstOrDefault(dir => System.IO.File.Exists(dir + "\\" + command));

            Console.WriteLine(
                (first != null)
                ? "command found at " + first + "\\" + command
                : "command not found");
        }
    }
}
